crates/sui-framework/packages/bridge/sources/bridge.move

```diff
@@ -540,363 +574,166 @@ module bridge::bridge {
-        let inner = load_inner_mut(&mut bridge);
-
-        // Assert the starting limit is a different value
-        assert!(limiter::get_route_limit(&inner.limiter, &chain_ids::get_route(chain_ids::eth_mainnet(), chain_ids::sui_mainnet())) != 1, 0);
-        // now shrink to 1 for SUI mainnet -> ETH mainnet
-        let msg = message::create_update_bridge_limit_message(
-            chain_ids::sui_mainnet(), // receiving_chain
-            0,
-            chain_ids::eth_mainnet(), // sending_chain
-            1,
-        );
-        let payload = message::extract_update_bridge_limit(&msg);
-        execute_update_bridge_limit(inner, payload);
-
-        // should be 1 now
-        assert_eq(limiter::get_route_limit(&inner.limiter, &chain_ids::get_route(chain_ids::eth_mainnet(), chain_ids::sui_mainnet())), 1);
-        // other routes are not impacted
-        assert!(limiter::get_route_limit(&inner.limiter, &chain_ids::get_route(chain_ids::eth_sepolia(), chain_ids::sui_testnet())) != 1, 0);
-
-        destroy(bridge);
-        test_scenario::end(scenario);
-    }
-
-    #[test]
-    #[expected_failure(abort_code = EUnexpectedChainID)]
-    fun test_execute_update_bridge_limit_abort_with_unexpected_chain_id() {
-        let mut scenario = test_scenario::begin(@0x0);
-        let ctx = test_scenario::ctx(&mut scenario);
-        let chain_id = chain_ids::sui_testnet();
-        let mut bridge = new_for_testing(ctx, chain_id);
-        let inner = load_inner_mut(&mut bridge);
-
-        // shrink to 1 for SUI mainnet -> ETH mainnet
-        let msg = message::create_update_bridge_limit_message(
-            chain_ids::sui_mainnet(), // receiving_chain
-            0,
-            chain_ids::eth_mainnet(), // sending_chain
-            1,
-        );
-        let payload = message::extract_update_bridge_limit(&msg);
-        // This abort because the receiving_chain (sui_mainnet) is not the same as the bridge's chain_id (sui_devnet)
-        execute_update_bridge_limit(inner, payload);
-
-        destroy(bridge);
-        test_scenario::end(scenario);
-    }
-
-
-    #[test]
-    fun test_execute_update_asset_price() {
-        let mut scenario = test_scenario::begin(@0x0);
-        let ctx = test_scenario::ctx(&mut scenario);
-        let chain_id = chain_ids::sui_testnet();
-        let mut bridge = new_for_testing(ctx, chain_id);
-        let inner = bridge.load_inner_mut();
-
-        // Assert the starting limit is a different value
-        assert!(inner.treasury.notional_value<BTC>() != 1_001_000_000, 0);
-        // now change it to 100_001_000
-        let msg = message::create_update_asset_price_message(
-            inner.treasury.token_id<BTC>(),
-            chain_ids::sui_mainnet(),
-            0,
-            1_001_000_000,
+    #[test_only]
+    public fun setup_treasury_for_testing(bridge: &mut Bridge) {
+        bridge.load_inner_mut().treasury.setup_for_testing();
+    }
```

Comment:
for these kind of test-only helpers, is it better to just make them `public(package` or `friend` (if it's still a thing in move 2024?) @dariorussi

Reply:
it does not really matter because those are compiled away.
I am not sure if there is a "standard" here, I'll find out.
We are talking about unit tests having visibility over the module they are testing but that is a change on the compiler side (and maybe more) that is not scheduled or planned yet

Reply:
cool, i'm curious too but not blocking this PR